tensorflow/stream_executor/device_description.cc

```diff
@@ -125,9 +125,9 @@ bool DeviceDescription::rocm_amdgpu_isa_version(int *version) const {
 
 bool ThreadDimOk(const DeviceDescription &device_description,
                  const ThreadDim &thread_dim) {
-  auto total_threads = thread_dim.x * thread_dim.y * thread_dim.z;
-  auto threads_per_block_limit = device_description.threads_per_block_limit();
-  if (total_threads > static_cast<uint64>(threads_per_block_limit)) {
+  int64 total_threads = thread_dim.x * thread_dim.y * thread_dim.z;
+  int64 threads_per_block_limit = device_description.threads_per_block_limit();
```

Comment:
`const`. Also, the previous cast was `uint64`, now these variables are `int64`. Do we miss the unsigned part?

Reply:
One was unsigned and the other was signed and thus I declare both as int64, as to not ever lose information.